### ISSUE TITLE
Suggest content for next MapLibre Newsletter

### ISSUE THREAD

**Issue by wipfli** [maplibre/maplibre.github.io#291]
If you would like to add something to the next MapLibre Newsletter, feel free to post a section here in this issue and then we will incorporate it in the next Newsletter.

**Comment by nyurik** [comment]
> Martin Tile Server v0.13 is out, adding support for tile cache.  Users can control how much memory to use for caching Postgres query results, as well as MBTiles and PMTiles access.  Note that compression results are not yet cached (see [#1112](https://github.com/maplibre/martin/issues/1112)), and neither are font and sprites.

P.S.  I think we should start this issue and link to it on every newsletter - this way people will know where to post things.

**Comment by nyurik** [comment]
A big discussion about on-boarding Maputnik is at https://github.com/maplibre/maplibre/issues/352

**Comment by acalcutt** [comment]
a new maplibre-native node version has been release, [node-v5.3.1](https://github.com/maplibre/maplibre-native/releases/tag/node-v5.3.1) which adds support for index-of and slice expressions, webp decoding support, and node 20 support.

Not sure this is newsletter worthy but I did have people asking for these features in node over the year.

**Comment by HarelM** [comment]
Version 4 of maplibre web has a release candidate with tons of improvements and should be released shortly, we are waiting for some feedback before we release it.
Sky spec was approved and should hopefully be incorporated in a future version, the original PR was updated.
Globe branch opened to allow PRs related to globe view.
A proposal to facilitate spec extension was made to the spec.

**Comment by wipfli** [comment]
Anything you would like to add to the Feb newsletter @HarelM @louwers ?

**Comment by HarelM** [comment]
Version 4.0.0, 4.0.1 and 4.0.2 were released and we are back with releasing versions more often after we finalized all the breaking changes in version 4.
We had an interesting discussion about geometry-type in the monthly meeting and we are looking for more feedback from the community around how to move forward:
https://github.com/maplibre/maplibre-style-spec/discussions/536
Sergei Bachinin was added as a contributor to maplibre-gl-js.

**Comment by louwers** [comment]
Yes, I will write it tomorrow.

**Comment by louwers** [comment]
## MapLibre Native

### Core

- Support for [multiple sprite sources](https://maplibre.org/maplibre-style-spec/sprite/#multiple-sprite-sources) is now available in the main branch and will be part of the next platform releases. Thanks to [@geolives](https://github.com/maplibre/maplibre-native/pull/1858) for their contribution.

### iOS

- After some trial and error, the release process for iOS is now fully automated. When the `VERSION` file and `CHANGELOG.md` are updated as part of a PR, the release workflow will automatically run. By getting changes in the hands of end-users quicker, we hope to encourage contributions for the iOS platform.
- The legacy `MGLStyleLayer` allowed using custom OpenGL ES code and shaders. A similar custom layer has been revived with the MapLibre Native iOS 6.1 release with `MLNCustomStyleLayer`, but now of course uses Metal.
- [New documentation for iOS](https://maplibre.org/maplibre-native/ios/latest/documentation/maplibre/) is available. It uses the [DocC](https://www.swift.org/documentation/docc/) documentation generator from Apple.

### Android

- The pre-release of MapLibre Native Android 11.0.0 is imminent. Those closely following the repository have already noticed it on [GitHub](https://github.com/maplibre/maplibre-native/releases/tag/android-v11.0.0-pre0). It will appear on [Maven Central](https://central.sonatype.com/artifact/org.maplibre.gl/android-sdk) shortly.
- For those interested, usage number for MapLibre Native for Android can be found below.

![maplibre_android_downloads](https://github.com/maplibre/maplibre.github.io/assets/649392/9ac56aef-3cf6-4592-b544-f749838b5ae8)

**Comment by HarelM** [comment]
Globe first PR was merged to a dedicated branch to foster collaboration.
Maplibre gl inspect was on boarded.
Versions 4.1.0 and 4.1.1 were released.
geometry-type was decided on and schedule for the next breaking change version.

**Comment by louwers** [comment]
- The MapLibre Native for Android 11.0.0 [pre-release](https://github.com/maplibre/maplibre-native/releases/tag/android-v11.0.0-pre0) has been out for almost a month now ([Maven Central](https://central.sonatype.com/artifact/org.maplibre.gl/android-sdk). We thank everyone for their feedback so far! Please give it a spin, and [report any issues](https://github.com/maplibre/maplibre-native/issues/1608) you encounter. The final release is set for the 28th of March if no critical issues are reported.

- We have additional [benchmarks](https://github.com/maplibre/maplibre-native/tree/main/benchmark) (using Google's Benchmark library) that are running on CI now. Keep an eye out for the results that show up on PRs. Contributors are invited to add additional benchmarks for performance critical code.

- Tim Sylvester added a PR to track what dependencies are used by expressions ([#2113](https://github.com/maplibre/maplibre-native/pull/2113)).

- Marc Wilson fixed an implementation issue with the location indicator on Android, which was causing threads to be constantly created and destroyed ([#2182](https://github.com/maplibre/maplibre-native/pull/2182)).

- A `#maplibre-swift` channel was created on Slack for discussion about a possible Swift-based iOS platform.

- Due to the switch to OpenGL ES 3.0 last year, builds for macOS from the `main` branch were temporarily not possible. Now that MapLibre Native has a Metal backend, macOS support can return on `main`. This was delayed a few times, but this month rendering for macOS was made possible again, by brining sample AppKit app back ([#2205](https://github.com/maplibre/maplibre-native/pull/2205)). Next GLFW will be made to run on macOS (useful for development purposes), and later Qt and Node.js are to return again macOS with Metal support. To everyone interested in using MapLibre Native on macOS, thanks for your patience!

![](https://files.slack.com/files-pri/T029HV94T-F06QT8K68BA/image.png)

**Comment by louwers** [comment]
Maybe another point:

- We have a new [release policy](https://github.com/maplibre/maplibre-native/wiki/Release-Policy-Android-and-iOS) concerning iOS and Android releases.

**Comment by louwers** [comment]
Metal-cpp comes with its own smart pointers. We were using them inappropiately in a few places, resulting in a memory leak. Resolved by [#2254](https://github.com/maplibre/maplibre-native/pull/2254) and [#2257](https://github.com/maplibre/maplibre-native/pull/2257).

We now use go through the entire iOS test app in an UI test to make sure nothing crashes as a smoke test ([#2258](https://github.com/maplibre/maplibre-native/pull/2258)). We also run that UI test with an undefined-behavior sanitizer on CI ([#2280](https://github.com/maplibre/maplibre-native/pull/2280)).

Some uniform buffer objects where moved to the global level, others to the layer level, resulting in fewer binds ([#2266](https://github.com/maplibre/maplibre-native/pull/2266) and [#2247](https://github.com/maplibre/maplibre-native/pull/2247) and [#2292](https://github.com/maplibre/maplibre-native/pull/2292)).

Allow changing User-Agent on iOS [#2293](https://github.com/maplibre/maplibre-native/pull/2293). This was the first pull request from [@hactar](https://github.com/hactar). 💪

MapLibre Native for Node.js 5.4.0 was [released](https://github.com/maplibre/maplibre-native/releases/tag/node-v5.4.0).

A (wide vector) shader was succesfully transplanted from another mapping toolkit ([#2183](https://github.com/maplibre/maplibre-native/pull/2183)). We want to make it possible to extend MapLibre Native. As you can see in the PR, we're not quite there that you can do this so without changing internals, but it is getting easier.

MapLibre Native is being rolled out in some larger deployment and several crashes have been [reported](https://github.com/maplibre/maplibre-native/issues?q=is%3Aissue+is%3Aopen+crash+created%3A%3E%3D2024-04-01+label%3Abug) this month. We now include debug symbols for iOS and Android in every release. If you observe a crash you can symbolicate the crash report to more easily identify where things went wrong.

Several reported issues with the [Android pre-release](https://github.com/maplibre/maplibre-native/issues/1608) have been fixed this month (e.g. [#2296](https://github.com/maplibre/maplibre-native/pull/2296)). We have a [milestone](https://github.com/maplibre/maplibre-native/milestone/16) where to-be-resolved issues can be tracked before the Android 11.0.0 release.

**Comment by HarelM** [comment]
# Web
Versions 4.1.2 and 4.1.3 were released this month.
We have recently merged a text fit feature by Microsoft that will be available in the next minor release that should allow keeping the aspect ratio of an image in some cases (Shields are obviously the main motivation here).
Globe branch is progressing nicely with the addition of fill-extrusion, hill-shade and lines.
Atmosphere is in discussion in terms of spec, and has an initial implementation that look really awesome!
![globe with atmosphere](https://github.com/maplibre/maplibre-gl-js/assets/2014432/b1b3e97f-beeb-4e07-addb-62b586411a88)


# Style Spec
The style spec docs were rewritten to use the awesome [Material for MkDocs](https://squidfunk.github.io/mkdocs-material/), which allows search and super easy maintenance. 

While the spec is not changing rapidly as we invest a lot of effort to keep it future ready on one hand and backward compatible on the other, making sure the docs are helpful is one of the most important aspects. This is vital so the community will be able to use it, and enjoy it.
We are looking for more expression examples in order to enrich the docs. This is probably the most complicated part of the spec, and we believe examples are very important to understand how things work.

The sky spec discussion has started looking at the atmosphere definitions as part of it, and there is a projection discussion for the globe.

A new type of source was proposed and approved: `contour`! This is the first time in a very long time a new type of source was approved! Very exciting!!

Feel free to join the discussions about the future of MapLibre!

**Comment by wipfli** [comment]
From @boldtrn https://github.com/maplibre/maplibre-navigation-ios/discussions/39#discussioncomment-9277657

We are currently reviving maplibre-navigation-ios. After we have improved maplibre-navigation-android recently, we are currently doing the same for iOS. We have updated dependencies like maplibre-native to version 6 with Metal support and use SPM instead of CocoaPods for dependency management. We have added hooks to make customizing some of the navigation functions possible. Kurviger is currently BEAT testing their app with maplibre-navigation-ios, if you would like to give it a spin, you can try the app for free [here](https://testflight.apple.com/join/Y2eMnS3D) - please note, only the navigation is using maplibre-navigation-ios. In the future we also want to make it easier to customize the out of the box navigation UI and to move the Android and iOS navigation libraries closer together. Would you like to join the effort? PRs and contributions are welcome.

**Comment by louwers** [comment]
## Newsletter May 2024 MapLibre Native

[MapLibre Native Android 11.0.0](https://github.com/maplibre/maplibre-native/releases/tag/android-v11.0.0) release. The [documentation](https://maplibre.org/maplibre-native/android/api/) was updated with the new package prefix.

 This month, several iOS patch releases were made as well as a pre-release for Android (e.g. [#2442](https://github.com/maplibre/maplibre-native/pull/2442), [#2395](https://github.com/maplibre/maplibre-native/pull/2395), [#2379](https://github.com/maplibre/maplibre-native/pull/2379)). More stability improvements are underway.

Stefan Karschti, one of the developers who implemented [Metal](https://maplibre.org/news/2024-01-19-metal-support-for-maplibre-native-ios-is-here/) support for MapLibre Native, has left the Metal team (a.k.a. MapLibre Native Team). Stefan, thank you for your many contributions! We are happy that you want to continue being a member of the MapLibre [community](https://maplibre.org/community/#8.49/46.7351/23.6051)!
 
One way to ensure MapLibre Native and MapLibre GL JS remain interoperable is to make sure we have the share the same render tests. Pulling the render tests in a shared repository would complicate pull requests too much. Instead, we wrote a script that we periodically run to check which render tests are missing in each repo. The result is a [render test parity status report](https://gist.github.com/louwers/0259c83872da7093670627a43f32b4a1) with corresponding [tracking issue](https://github.com/maplibre/maplibre-native/issues/2445).

Some guides have been added to the [iOS Documentation](https://maplibre.org/maplibre-native/ios/latest/documentation/maplibre/) demonstrating how to use the library with SwiftUI. People who are familiar with using MapLibre Native on iOS, are invited to make a PR to [add more guides](https://github.com/maplibre/maplibre-native/blob/main/platform/ios/CONTRIBUTING.md#documentation) to the DocC-based documentation site.

**Are you using MapLibre Native?** Please leave a comment on the [discussion thread](https://github.com/maplibre/maplibre-native/discussions/690) on GitHub.

**Comment by HarelM** [comment]
# Web
We have released four (!) versions this month: 4.2.0, 4.3.0, 4.3.1, 4.3.2.
We finally added support for the `distance` expression that was missing in terms of parity between web and native, thus making web a superset of native when it comes to expression support - meaning that all the styles that work for native should work for web as well and look similar. Thanks to @louwers we now can properly track this with the parity script. We also added to the style spec docs a reference to issues that prevent parity to allow you to see where it is standing and help out.

As part of the globe effort there was a need to fix the collision boxes bugs. Jakub, who is making a great progress with the globe code, had been able to port these fixes to the main version, making the collision boxes experience a lot better.

The globe is in its final stretch, most of the issues have been fixed and most of map features and style are supported.
There are still work to be done but you can admire how nice it looks now:
![Screenshot of a globe with symbols and debug collisions, with many test texts.](https://github.com/maplibre/maplibre-gl-js/assets/57600346/1e19d7be-43da-4395-a053-10a341b988c4)

There are some minor changes with rendering we expect to introduce in the next version related to brackets and new lines so keep an eye for those.

We've fixed most of the issues with movement when 3D terrain is on so now the panning and pinching is a lot smoother.

Overall, this month was amazing in terms of contribution from the community, and I think the library is steadily gaining momentum, so we would like to thank the community for upstreaming their work!

**Comment by nyurik** [comment]
> The next generation tile format (MLT) efforts have gained a lot of speed with Microsoft sponsoring initial proof of concept. All work is being done in the recently on-boarded [maplibre-tile-spec](https://github.com/maplibre/maplibre-tile-spec) repository from @mactrem  who did extensive research in the topic. We already get up to 6x size reduction over MVT tiles for large tiles, and plan a lot of improvements including zero-copy direct to GPU pre-tessellated geometries and ability to specify sub-geometry's range values and other data types.


The above can be edited a bit for content and styling

**Comment by ebrelsford** [comment]
@nyurik A few suggestions:
 * Can we mention that Stamen is assisting with this work?
 * Maybe it would be helpful to point people to the slack channel to follow along?
 * It might be worth managing expectations, the proof of concept target is to benchmark parsing MVTs and MLTs in JS to get a sense of the client side impact of switching.

**Comment by mactrem** [comment]
@nyurik i would also suggest, since it is frequently discussed,to add that Mlt has support for a more complex  type system such as nested types or lists as well as linear referencing and m-values

**Comment by wipfli** [comment]
Thanks for the feedback, I will try to incorporate those. If I miss something, there will always be a chance to mention more in the June newsletter...

**Comment by nyurik** [comment]
:tada: Long awaited Martin tile server v0.14 has been released!
* AWS Lambda support
* Options to set preferred response encodings (e.g. if browser supports gzip & brotli, can set which should be returned)
* `mbtiles` can now create binary-diffing between tiles, making diffs much smaller
* numerous bug fixes, docs, and other improvements

https://github.com/maplibre/martin/releases/tag/v0.14.0

**Comment by HarelM** [comment]
## Web

Two major version were released: 4.4.0 and 4.5.0.
Some of the highlights of these version are the addition of a new unminified production build and an initial sky implementation that includes sky color, horizon color and fog color:
![sky](https://github.com/maplibre/maplibre-gl-js/assets/3269297/23051426-0495-4981-aa92-9a907aea663b)

This sky PR was waiting for a long time for some love and care and with the help of Jakub and Vivian we were finally able to push it through, thanks guys!

Atmosphere was added to the globe branch to make the globe more realistic.
![atmosphere](https://github.com/maplibre/maplibre-gl-js/assets/2014432/b1b3e97f-beeb-4e07-addb-62b586411a88)

## Style spec

Sky spec was updated to allow moving forward with the sky implementation and atmosphere, the sky spec is still experimental though.

**Comment by louwers** [comment]
## MapLibre Native


This month the MapLibre Native (Rendering) Team, previously known as the Metal team, kicked off the R&D for a Vulkan backend.

[Vulkan](https://en.wikipedia.org/wiki/Vulkan) is a next-generation graphics API developed by the Khronos Group, the same entity behind OpenGL (ES). While OpenGL ES has served us well, and will continue to do so for the forseeable future, it was developed a long time ago and it is beginning to show its age. We are running into some of its limitations when trying to realize further performance gains.

While a Vulkan backend will likely not give us an automatic performance boost, it does offer a lot more control. We know that the automotive industry is following this development with interest. It turns out that in-car systems tend to use resource-constrained customized SOCs where Vulkan offers exactly the kind of tuning and observability that is helpful. Of course, phones, desktops and other devices that support Vulkan, where MapLibre Native is already commonly used, also stand to benefit.

[Adrian Cojocaru](https://github.com/adrian-cojocaru) joined the MapLibre Native (Rendering) Team and the MapLibre community this month. A warm welcome to Adrian! He is a graphics engineer with a background in the games industry and is bolstering the Vulkan expertise of the team. He has hit the ground running this month and already shared a screenshot with a functional fill layer that uses Vulkan.

![image (1)](https://github.com/maplibre/maplibre.github.io/assets/649392/d76d3ca3-c418-4c11-8592-4d15f15bacee)


If you want to stay up-to-date, join the discussion or even participate in the development of the Vulkan backend, join us on GitHub, the monthly TSC meetings and/or on Slack.

[MapLibre Android 11.0.1](https://github.com/maplibre/maplibre-native/releases/tag/android-v11.0.1) was released.

[MapLibre Native iOS 6.5.0](https://github.com/maplibre/maplibre-native/releases/tag/ios-v6.5.0) was released.

Building an app using MapLibre Native with just the API documentation and snippets of code scattered over the internet or in for example the test apps, is not an easy task. We are working to consolidate existing examples and write new ones for our documentation.  We created milestones to track progress on this (see [iOS Examples](https://github.com/maplibre/maplibre-native/milestone/15) and [Android Examples](https://github.com/maplibre/maplibre-native/milestone/17)). If you have interesting usage patterns to share, you are invited to create an issue or make a pull request.

## Events

- FOSS4G EU, 1-7 July, Tartu (Estonia)
- State of the Map Europe, 18-21 July, Łódź (Poland)
- FOSS4G BE+NL, 25-27 September 2024, Baarle (Belgium & the Netherlands)

**Comment by ebrelsford** [comment]
## maplibre-tile-spec (MLT)

Stamen is wrapping up work on [the proof of concept MLT JavaScript decoder](https://github.com/maplibre/maplibre-tile-spec/). The JavaScript decoder is at parity with the Java decoder. See the repo for an experimental JS decoder package and initial benchmarks (against MVT parsing). You can also see an [experiment](https://mltdemo.stamen.com/) rendering Bing Maps-based MLTs with MapLibre GL JS:

![image](https://github.com/maplibre/maplibre.github.io/assets/375002/232c2f26-79e5-4a23-b71e-1734b902d3b9)

**Comment by louwers** [comment]
Adrian has made a Vulkan Draft PR now to track progress, you can use that image instead and link to the PR. https://github.com/maplibre/maplibre-native/pull/2564

**Comment by HarelM** [comment]
## Web

The main focus of this month was to push forward the globe effort, where the main focus now is on globe controls, gestures, and refactoring the transform class to be different between globe and mercator view - this is the final stretch for the globe and we are aiming to release it by the end of this year. I would like to repeat my gratitude to Jakub for his amazing work here on the globe implementation.
While the changes in globe are not considered breaking changes the direction right now is to release it under version 5 to avoid unexpected breaking changes in plugins and other external libraries that might expect certain behavior from internal implementation such as the transform class.

**Comment by louwers** [comment]
## MapLibre Native

- The initial [Vulkan backend](https://github.com/maplibre/maplibre-native/pull/2564) implementation is almost ready to merge! In one month it went from a proof-of-concept to passing all render tests. We don't have any performance numbers at this point, and it is probably too early for those anyway. To fully exploit the capabilities of Vulkan and use it to tune performance we may need to make some architectural challenges. We still need to figure out how to intergrate it with Android and how to distribute it for that platform. Of course, OpenGL ES will still be supported as well. That said, impressive and promising progress was made this month.  
- We now run the C++ Unit Tests on Android as part of CI.
- Some new articles have been added to the new iOS documentation, including information on the [snapshotter](https://maplibre.org/maplibre-native/ios/latest/documentation/maplibre-native-for-ios/staticsnapshotexample), [customizing fonts](https://maplibre.org/maplibre-native/ios/latest/documentation/maplibre-native-for-ios/customizing_fonts), [Info.plist keys](https://maplibre.org/maplibre-native/ios/latest/documentation/maplibre-native-for-ios/info.plist_keys), [user interactions](https://maplibre.org/maplibre-native/ios/latest/documentation/maplibre-native-for-ios/gesturerecognizers), [adding multiple images](https://maplibre.org/maplibre-native/ios/latest/documentation/maplibre-native-for-ios/multipleimagesexample) and [offline](https://maplibre.org/maplibre-native/ios/latest/documentation/maplibre-native-for-ios/offlinepackexample) [capabilities](https://maplibre.org/maplibre-native/ios/latest/documentation/maplibre-native-for-ios/manageofflineregionsexample).
- [Compose Multiplatform](https://www.jetbrains.com/lp/compose-multiplatform/) is a cross-platform UI software development kit from Jetbrains (somewhat similar to Flutter, but using Kotlin instead of Dart). There is some interest from the community to develop a MapLibre Native library for Compose Multiplatform (not to be confused for Jetpack Compose, for which libraries already exist). Please see the [issue](https://github.com/maplibre/maplibre-native/issues/2638) for a discussion.
- [MapLibre Android 11.1.0](https://github.com/maplibre/maplibre-native/releases/tag/android-v11.1.0) was released.
- [MapLibre iOS 6.5.2](https://github.com/maplibre/maplibre-native/releases/tag/ios-v6.5.2) was released.
- Some improvements were made to the included [Dockerfile](https://github.com/maplibre/maplibre-native/tree/main/docker) and accompanying instructions to make Linux builds of MapLibre Native.
- A PR showing a [proof-of-concept](https://github.com/maplibre/maplibre-native/pull/2647) that intergrates Rust build tools into the CMake build config was opened this month. The PR swaps out [cppcolorparser](https://github.com/maplibre/maplibre-native/blob/main/vendor/csscolorparser/csscolorparser.cpp) with a similar Rust library, just to see if a Rust dependency can be added while still building and running. It does build, and it does run.

**Comment by louwers** [comment]
## MapLibre Native

- An initial intergration of the Vulkan backend with Android is expected to land soon ([#2711](https://github.com/maplibre/maplibre-native/pull/2711)). We will be releasing an 'Early Access' version of MapLibre Android which will contain only Vulkan support next month (September 2024). This version is not intended to be used in production, but a way to get the Vulkan based renderer in the hands of the community to get feedback. If you are interested to help testing Vulkan on Android please leave a commend in [this GitHub thread](https://github.com/maplibre/maplibre-native/issues/2787) and we will notify you once the release is out.
- We added back a CMake configuration for Metal builds on macOS. Thanks to this, the Node.js platform is now using the `main` branch again instead of the `opengl-2` branch. A [pre-release for MapLibre Node.js 6.0.0](https://github.com/maplibre/maplibre-native/releases/tag/node-v6.0.0-pre.1) with Metal support on macOS is out, with the full release immanent.
- This is a Public Service Announcement for everyone using **MapLibre iOS with Annotations**. Some users reported [synchronization issues](https://github.com/maplibre/maplibre-native/issues/2522) when panning the map. The issue is only present on devices with ProMotion (120Hz) displays and can be fixed by updating the `Info.plist` for your app (see [Apple documentation](https://developer.apple.com/documentation/quartzcore/optimizing_promotion_refresh_rates_for_iphone_13_pro_and_ipad_pro?language=objc)).
- We started using C++20 this month. 
- We forked mapbox-gestures-android and published it as a new package [maplibre-gestures-android](https://github.com/maplibre/maplibre-gestures-android).
- All features of the [MapLibre Style Spec](https://maplibre.org/maplibre-style-spec/) that are supported by MapLibre GL JS but not (yet) by MapLibre Native now have corresponding [tracking issues](https://github.com/maplibre/maplibre-native/issues?q=is%3Aissue+is%3Aopen+label%3Ajs-parity).
- We are using the latest version of the [style spec](https://github.com/maplibre/maplibre-native/blob/main/scripts/style-spec-reference/v8.json) that we share with MapLibre GL JS again. Of course we need to apply a few patches for yet unsupported features.
- [MapLibre Android 11.2.0](https://github.com/maplibre/maplibre-native/releases/tag/android-v11.2.0) was released
- [MapLibre iOS 6.5.4](https://github.com/maplibre/maplibre-native/releases/tag/ios-v6.5.4) was released.

**Comment by HarelM** [comment]
## Web

Three versions were released this month: 4.5.1, 4.5.2 and 4.6.0.
Among the highlight of these versions are a lot of small bug fixes, better CJKV characters handling and better handing of the camera when terrain is enabld.
Globe branch now supports custom layer interface and trasform was broken down there. 
When globe branch will be merged, which should happen in the next month or two we will start preparing version 5 with its breaking changes as can be seen [here](https://github.com/maplibre/maplibre-gl-js/issues/3834).

## Maputnik

Maputnik, our official style editor, just got a major new ability to be traslated to multiple languages. It already support English, Japanese, Chinese and Hebrew! If you'd like to contribute a translation we would appreciate it a lot!

## Maplibre Geocoder

Maplibre geocoder, our small search component for maplibre-gl-js got a facelift with the migration to typescript, jest and typedoc

**Comment by birkskyum** [comment]
Just so i don't forget about it - the much improved python support gl js has through the new plotly release.

**Comment by ianthetechie** [comment]
# SwiftUI DSL

MapLibre has a new hosted project: the [SwiftUI DSL for MapLibre native](https://github.com/maplibre/swiftui-dsl)! Originally started by Ian Wagner from [Stadia Maps](https://stadiamaps.com/), the goal of the project is to give MapLibre developers on iOS an experience that rivals the latest MapKit for SwiftUI APIs.

[Jacob Fielding](https://github.com/archdoog) ([Rallista](https://rallista.app/)), [Patrick Wolowicz](https://subzero.eu/) and [Patrick Kladek](https://github.com/Patrick-Kladek) (both working with support from [HudHud](https://hudhud.sa/en)) shared the same vision, and have each contributed significantly to the ongoing development over the last year. This is a great example of the collaboration that MapLibre seeks to enable across the industry, and the authors are excited to have the project officially hosted under the MapLibre organization.

If you'd like to help shape the future of MapLibre in SwiftUI, join us in the `#maplibre-swiftui-compose-playground` channel on [Slack](https://slack.openstreetmap.us/).

**Comment by louwers** [comment]
## MapLibre Native

- The most significant development this month was the [MapLibre Native Android pre-release](https://github.com/maplibre/maplibre-native/releases/tag/android-v11.3.0-vulkan-pre0) that uses [Vulkan](https://www.vulkan.org/) for rendering. We did some benchmarks and the results [look promising](https://github.com/maplibre/maplibre-native/issues/2787#issuecomment-2368938676). Especially newer devices seem to benefit from this modern graphics API, although we are still validating these results and exploring opportunities to take advantage of Vulkan's capabilities. Since the Vulkan backend is still in development, now is the best time to try it out and to let us know any regressions you encounter (either as an issue or in the [pre-release thread](https://github.com/maplibre/maplibre-native/issues/2787)). You can test it by using `11.3.0-vulkan-pre0` as a version (or the latest Vulkan version from [Maven Central](https://central.sonatype.com/artifact/org.maplibre.gl/android-sdk/versions)).

- Support for [`textFitWidth` and `textFitHeight`](https://maplibre.org/maplibre-style-spec/sprite/#text-fit-properties) properties landed in [#2780](https://github.com/maplibre/maplibre-native/pull/2780) this month. Support for specifying [padding for icons](https://maplibre.org/maplibre-style-spec/types/#padding) on all sides separately landed in [#2845](https://github.com/maplibre/maplibre-native/pull/2845). Both features are used in Bing Maps, and were implemented by Microsoft engineers (among which Alexey Kon). We are thankful for these contributions that work towards [feature parity with MapLibre GL JS](https://github.com/maplibre/maplibre-native/issues?q=is%3Aissue+is%3Aopen+label%3Ajs-parity).

- MapLibre Native Android [11.3.0](https://github.com/maplibre/maplibre-native/releases/tag/android-v11.3.0), [11.4.0](https://github.com/maplibre/maplibre-native/releases/tag/android-v11.4.0) and [11.5.0](https://github.com/maplibre/maplibre-native/releases/tag/android-v11.5.0) were released. This last release inadvertently included a small API change, which will mostly be resolved again by [#2880](https://github.com/maplibre/maplibre-native/pull/2880#pullrequestreview-2336900203). A pesky bug in the emulator was plaguing MapLibre Native Android ever since the 11.x.x, but we added a workaround in the latest release, so that working with the emulator should be viable again.

- MapLibre Native iOS [6.6.0](https://github.com/maplibre/maplibre-native/releases/tag/ios-v6.6.0) and [6.7.0](https://github.com/maplibre/maplibre-native/releases/tag/ios-v6.7.0) were released.

- MapLibre Node.js 6.0.0 was released which brings support for Metal on macOS, and uses the new [drawable](https://github.com/maplibre/maplibre-native/blob/main/design-proposals/2022-10-27-rendering-modularization.md) renderer architecture.

- Support for [PMTiles](https://github.com/protomaps/PMTiles) may come to MapLibre Native soon! There is an open PR ([#2882](https://github.com/maplibre/maplibre-native/pull/2882)) from Tiago Costa with a request to try it out.

**Comment by louwers** [comment]
## MapLibre Native

- Development of the Vulkan backend continues. We ran another benchmark on AWS Device Farm, using even more device types this time. One conclusion you may draw from the [results](https://github.com/maplibre/maplibre-native/issues/2787#issuecomment-2432251912) is that Vulkan will offer a solid performance boost on modern devices. Stay tuned for an Android official release following later this year!
- Kaushal Kumar Singh rewrote the Android build config from Groovy to the more modern Kotlin in [#2902](https://github.com/maplibre/maplibre-native/pull/2902).
- Support for [`text-variable-anchor-offset`](https://maplibre.org/maplibre-style-spec/layers/#text-variable-anchor-offset) will be merged soon ([#2921](https://github.com/maplibre/maplibre-native/pull/2921)). This feature is one of the features that Microsoft championed to be added to MapLibre GL JS, and now ported to MapLibre Native. We're happy with this contribution and are excited about Microsoft adopting MapLibre Native.
- [MapLibre iOS 6.7.1](https://github.com/maplibre/maplibre-native/releases?q=ios&expanded=true) was released.
- [MapLibre Android 11.5.1 and 11.5.2](https://github.com/maplibre/maplibre-native/releases?q=android-v11&expanded=true) were released.
- We also made two releases for [MapLibre Android v10](https://github.com/maplibre/maplibre-native/releases?q=android-v10&expanded=true) with some backports.

**Comment by josxha** [comment]
## Flutter MapLibre GL

We’re conducting a brief survey to gather feedback on MapLibre Flutter. The survey takes only about 2 minutes to complete and will help to identify strengths and areas for improvement. If you've worked with MapLibre on Flutter, your input is highly appreciated.

You can participate here: https://forms.gle/UTB6KjVFtmYv2RzU8

**Comment by HarelM** [comment]
## Web

We have released versions 5.0.0-pre.2 and 5.0.0-pre.3 this month.
We have introduced some breaking changes in these versions as was planned for version 5.
There are still open issues with globe and other improvement we plan to add as part of this major release.
See the initial post in the following issue to see the current status:
- https://github.com/maplibre/maplibre-gl-js/issues/3834

Overall we are progressing well and ironing out the globe feature.

**Comment by ibesora** [comment]
I have merged a fix for raster layers when using globe and terrain [here](https://github.com/maplibre/maplibre-gl-js/pull/4825) but I think that a sentence saying we are ironing out the globe feature would be enough

**Comment by louwers** [comment]
**Events:**

- [FOSS4G 2025](https://2024.foss4g.org/en/) Belem
- [FOSSGIS 2025](https://fossgis-konferenz.de/2025/programm/) Münster (German)

---

Maybe we can put a link to this thread to ask people to contribute content for the next newsletter.

---

Maybe we can also put times & links to the TSC meetings in November.

**Comment by meenaxc** [comment]
> **Events:**
> 
> * [FOSS4G 2025](https://2024.foss4g.org/en/) Belem
> * [FOSSGIS 2025](https://fossgis-konferenz.de/2025/programm/) Münster (German)
> 
> Maybe we can put a link to this thread to ask people to contribute content for the next newsletter.
> 
> Maybe we can also put times & links to the TSC meetings in November.

Hi @ramyaragupathy when are the TSC and/or monthly meetings taking place?

**Comment by louwers** [comment]
## MapLibre Native

- **MapLibre Android** [11.6.0](https://github.com/maplibre/maplibre-native/releases/tag/android-v11.6.0) and [11.6.1](https://github.com/maplibre/maplibre-native/releases/tag/android-v11.6.1) have been released.
- **MapLibre iOS** [6.8.1](https://github.com/maplibre/maplibre-native/releases/tag/ios-v6.8.1) was released.
- A new documentation site for [MapLibre Android Examples](https://maplibre.org/maplibre-native/android/examples/) has been launched! This provides a more accessible resource compared to the [API Documentation](https://maplibre.org/maplibre-native/android/api/). More examples are in progress, and contributions are warmly welcomed.
- [Flitsmeister](https://www.flitsmeister.nl/) has released a sample project demonstrating how to use MapLibre Android with **Android Auto**: [GitHub Repo](https://github.com/flitsmeister/MapLibre-Android-Auto-Sample).  We're thrilled to see Flitsmeister share their hard-earned expertise with the community. It's also exciting to see the FOSS magic at work, with suggestions for improvements already coming from other contributors.
- We're experimenting with some new GitHub functionality: [issue templates and issue types](https://github.com/maplibre/maplibre-native/discussions/3053).
- Engineers working for HudHud have kicked off the development of brining **3D Model Support** to MapLibre Native. To implement this properly we need to "revive the 3D pass" (which is currently unused), which brings about plenty of complexities. It will be a while until proper 3D support will land, but watch this space, and join the monthly meetings you are interested in this.
### Vulkan

We’ve made a **pre-release** of MapLibre Android with **Vulkan support**: [11.6.1-vulkan-pre1](https://github.com/maplibre/maplibre-native/releases/tag/android-v11.6.1-vulkan-pre1). This will likely be the last pre-release, with the official Vulkan-supported release planned for release in early December.

While OpenGL ES will remain supported and MapLibre Android releases with OpenGL ES will continue for the foreseeable future, Vulkan offers a more modern and performant backend.

#### Feedback Needed!

Help us test Vulkan support and ensure there are no regressions for your app by following these steps:

1. Use the dependency: `org.maplibre.gl:android-sdk:11.6.1-vulkan-pre1`.
2. Test your styles:
    - Verify they render correctly across various zoom levels, pitches, and feature-dense areas.
    - Rotate the device and check for any issues.
    - Ensure there are no performance regressions.
    - Test runtime styling, especially if it’s user-driven.
3. Share your feedback! You can contribute to the [pre-release thread](https://github.com/maplibre/maplibre-native/issues/2787) or [create an issue](https://github.com/maplibre/maplibre-native/issues/new?template=android-bug-report.yml). Positive feedback is welcome, too!

### Kotlin Multiplatform

Exciting news for developers interested in [**Kotlin Multiplatform**](https://kotlinlang.org/docs/multiplatform.html): [Sargun Vohra](https://github.com/sargunv), with input from [Tobias Zwick](https://github.com/westnordost), has released [MapLibre for Compose](https://github.com/sargunv/maplibre-compose#maplibre-for-compose)!

The library integrates MapLibre iOS and MapLibre Android and is open to contributions. The README lists various ideas, including adding support for web (presumably by intergrating MapLibre GL JS), which would make the library truly cross-platform. Very cool to see this project take shape.


<p float="left">
  <img src="https://github.com/user-attachments/assets/671fc56d-b1a8-4843-82ea-4813c9712ea3" height=400/>
  <img src="https://github.com/user-attachments/assets/ae4a43e4-66bc-4a69-824f-957874912abd" height=400/>
</p>

**Comment by HarelM** [comment]
## Web

We have released versions 5.0.0-pre.5, 5.0.0-pre.6, 5.0.0-pre.7 and 5.0.0-pre.8 this month!
These version introduced most of the breaking changes we have planned for this version.
For other feature we thought about introducing as a breaking change, we found a solution that does not require a breaking change and we'll see if and when they will be introduced in the future.
We have solved all the showstopper bugs for this release and there's only a few minor things we would like to include before creating a production ready version.
To see the current status of version 5 you can follow the initial post in the following issue:
- https://github.com/maplibre/maplibre-gl-js/issues/3834

**Comment by josxha** [comment]
## Flutter MapLibre GL

The survey conducted last month found that users appreciate MapLibre on Flutter and rate it 4.1 out of 5 stars overall. The package is praised for its performance, lower cost and integration with Flutter, among other things.
Stability and platform differences were identified as issues, as 46% of users reported have been affected by platform-specific behavior and 44% of users experienced crashes.

<img src="https://github.com/user-attachments/assets/c1ae5448-8e94-45dd-8687-6292329d7210" width="30%" />

**Comment by boldtrn** [comment]
## Navigation Android

Maplibre-Navigation-Android has been updated to use the latest Maplibre-Native release 11.6.1 and was released as version 4.0.0 with this version. You can find more information about this in the following PR:

- https://github.com/maplibre/maplibre-navigation-android/pull/110

**Comment by louwers** [comment]
@ramyaragupathy My note about Compose Multiplatform did not make it to the newsletter. Could you include it in the December newsletter?

**Comment by sargunv** [comment]
MapLibre Compose (Multiplatform) links/info:

* Originally released late November, now on v0.3.0. It wasn't in the November newsletter so y'all can announce it as a new project in the December newsletter.
* Repo: https://github.com/sargunv/maplibre-compose
* Documentation: https://sargunv.github.io/maplibre-compose/
* Current screenshots from readme:

<p float="left">
  <img src="https://github.com/user-attachments/assets/997cf8a4-2841-40c8-b5a1-ef98193b21b2" width="200"/>
  <img src="https://github.com/user-attachments/assets/e450f689-e254-48b7-bd91-3d3042faa290" width="200"/>
</p>

**Comment by CommanderStorm** [comment]
# Martin

Martin Tile Server `v0.15.0` will have been released™:
- https://github.com/maplibre/martin/issues/1632

While there are a [lot of changes](https://github.com/maplibre/martin/releases/tag/v0.15.0), these changes stand out:
- Basic [`cog` (Cloud Optimized GeoTIFF) support](https://maplibre.org/martin/sources-cog-files.html) was added as a data source. This allows to effectively serve raster tiles similar to mbtiles raster mode and pmtiles raster mode. See [our docs](https://maplibre.org/martin/sources-cog-files.html) for further context.
- `SDF` sprites can now be generated via a separate API route. This allows to color sprites in `maplibre-gl`/`maplibre-native` based on map data and have crispier sprites.
  See [our docs](https://maplibre.org/martin/sources-sprites.html) for further context.
  While this seems like an obvious better choice, these icons are not clearly supperiour as they :
  - only support one color
  - don't support configuring the background other than a blurred border.
 
  Results of this are for example

  |  |  |
  |--------|--------|
  | Example | ![Image](https://github.com/user-attachments/assets/995613bc-77aa-4ecc-975f-496d6561d884) |
  | Where is this from? | [NavigaTUM](https://nav.tum.de) |

- We have implemented the basic framework on how to add a web-UI.
  In the current state, this is not quite usable, but we are getting coser ^^

Open questions:

@hiddewie (since you already have expermeinted with this): do you have another example of sdf sprites?
@sharkAndshark do you maybe have a better example of how to present COGs? What I wrote seems a bit.. bland

**Comment by louwers** [comment]
We already had the Vulkan release announcement, which we can remind people of in the newsletter. 

## MapLibre for Compose

Please combine [my comment](https://github.com/maplibre/maplibre.github.io/issues/291#issuecomment-2505051503) (Kotlin Multiplatform) and [Sargun's comment](https://github.com/maplibre/maplibre.github.io/issues/291#issuecomment-2558201642) to announce Kotlin Multiplatform as a new platform. 🙂 

## MapLibre Native

- MapLibre Android [11.7.0](https://github.com/maplibre/maplibre-native/releases/tag/android-v11.7.0) and [11.7.1](https://github.com/maplibre/maplibre-native/releases/tag/android-v11.7.1) were released. The first releases which are available with Vulkan support (see [announcement](https://maplibre.org/news/2024-12-12-maplibre-android-vulkan/)).
- MapLibre iOS [6.9.0](https://github.com/maplibre/maplibre-native/releases/tag/ios-v6.9.0) was released.

**Comment by josxha** [comment]
## Flutter MapLibre

Started in September 2024, the Flutter maplibre package was created to provide modern Flutter bindings for MapLibre Native and MapLibre GL JS and a seamless integration into the Flutter ecosystem. Currently, supported platforms are Android and Web, while the iOS implementation is on the way and, as of right now, about halfway done (see [#26](https://github.com/josxha/flutter-maplibre/issues/26)).

The package already receives a couple of hundred downloads each month, and its goal is to become a full alternative to flutter-maplibre-gl by the end of Q1 2025.

- Repository: [github.com/josxha/flutter-maplibre](https://github.com/josxha/flutter-maplibre)
- pub.dev Package Registry: [pub.dev/packages/maplibre](https://pub.dev/packages/maplibre)
- Hosted Example App: [flutter-maplibre.pages.dev/demo](https://flutter-maplibre.pages.dev/demo/)

**Comment by birkskyum** [comment]
> modern Flutter bindings

@josxha , nice project - a small request from those of use not too regular users of flutter - I think it would be a good idea to add a "[Why](https://vite.dev/guide/why.html) Flutter-MapLibre?" or a "[Comparison](https://vite.dev/guide/comparisons.html) to alternatives" page in the docs, similar to the vite pages I linked, to emphasize the motivation/benefits of this new binding. I have some guesses what's meant by this binding being more "modern", but I think it could help adoption to emphasize the benefits.

**Comment by josxha** [comment]
Thanks for your feedback @birkskyum, that's a great idea! I've already had some basic feature comparison in the docs but now added additional informations. https://flutter-maplibre.pages.dev/compare/ Let me know if it goes the right direction. As the text related to maplibre_gl is relatively long, I assume linking it would be best?

**Comment by louwers** [comment]
@josxha Adding a link sounds good to me. Thanks for sharing, looks like an interesting project!

**Comment by birkskyum** [comment]
The **MapLibre Native - React Native SDK v10** has been [released](https://github.com/maplibre/maplibre-react-native/releases/tag/v10.0.0), and I think it definitely deserves a mention.


It would be great to have a section about this release in the newsletter. These are some of the changes I find particularly awesome:

_NOTES/DRAFTING BELOW_

- New Architecture - The SDK hasn't been refactored to use New Architecture yet, so there isn't any performance improvement, but v10 does contain a new compat layer that allow MapLibre Native to be used in New Architecture mode.
- Expo - The v10 does have improvements for use in Expo, such as the ability to add Google Location Services.
- This release allow users to configure which version of the iOS and Android SDK's from the MapLibre Native distributions that are used. This will work at least for minor and patch releases, and it allow users of MapLibre React Native to adopt these new versions immediately after they're published, without the delay of a MapLibre React Native release.
- New Documentation: https://maplibre.org/maplibre-react-native/
- This release has been a year in the making, with 29 alphas, and 21 betas.

cc @tyrauber, @kiwikilian - Feel to help improve the content

**Comment by ianthetechie** [comment]
The MapLibre SwiftUI DSL v0.5 release is out with two stand-out features:

* Basic support for fill style layers in the DSL. This lets you add things like area of interest shading dynamically using the DSL.
* A revamped read-only proxy interface to the map view. This lets us safely expose access to functions on the underlying map view like `func convert(_ coordinate: CLLocationCoordinate2D, toPointTo: UIView?) -> CGPoint` directly. Such functions won't cause any interference with SwiftUI state, and we now have a way to expose them directly, unlocking even more use cases.

Both new features were implemented by @hactar. These are first passes and don't expose everything theoretically possible by MapLibre. If you'd like access to some method on the `MLNMapView` or need more properties from `MLNFillStyleLayer`, [let us know with an issue or open a PR on GitHub](https://github.com/maplibre/swiftui-dsl)!

We've also done some dependency upgrades, including upgrading MapLibre Native to 6.10. @Archdoog notes that if you're using the Mockable package, the change may require some updates to your own code.

**Comment by bdon** [comment]
# MapLibre Web Fonts

A group of developers using MapLibre GL JS are interested in building native Web Fonts (`.woff2`) support, as a replacement for needing to fetch and host `glyphs`. This would have key advantages of:

* Fewer font files and less data to download when displaying labels
* Easier workflow for designing maps with custom typography
* Support for more languages in South and Southeast Asia

You can find more information on [GitHub Discussions](https://github.com/maplibre/maplibre-gl-js/discussions/2555) or in the `#maplibre-i18n` slack channel.

**Comment by josxha** [comment]
## MapLibre Flutter

- The MapLibre Native Android SDKhas been updated to 11.8.0 which adds support for PMTiles. Support on web is planned to provide feature parity across all platforms.
- Static typed map styles including expressions have been evaluated. Modelling the maplibre style spec with JSON schema to make the code generation more straight forward looks promising.

**Comment by HarelM** [comment]
## Web

The year started with a bang. With the release of the globe almost at new year's eve we were off to a great start of 2025. A very important discussion started with the release of version 5 and the change we decided to do with `geometry-type`. Which drive the trust of 5.0.1 soon after and the revert of a breaking change in the style. 
While this discussion was valuable we do want to encourage the community to use our pre-releases for majors version and get that feedback before the official version is released. 
We also released version 5.1.0 this month, and there are nice things cooking as can be read in this newsletter. Great year ahead, no doubt!

**Comment by nyurik** [comment]
* MapLibre Native gained experimental Rust support ([docs](https://maplibre.org/maplibre-native/docs/book/rust.html), [PR](https://github.com/maplibre/maplibre-native/pull/3137)). This allows C++ code to use Rust modules including dependencies. The first demo replaces csscolorparser.hpp with a popular [Rust lib](https://crates.io/crates/csscolorparser) (2.2 million downloads). We need feedback on any potential issues with this approach. Thanks to @louwers and @nyurik 
* Work [has began](https://github.com/maplibre/maplibre-native/pull/3181) on MapLibre Native Rust wrapper - ability to use native lib as a Rust crate. This will add another supported platform, in addition to Linux/Android/iOS/Node/...

**Comment by louwers** [comment]
## MapLibre Native 

- [MapLibre Android 11.8.0](https://github.com/maplibre/maplibre-native/releases/tag/android-v11.8.0) and [MapLibre iOS 6.10.0 ](https://github.com/maplibre/maplibre-native/releases/tag/ios-v6.10.0) were released which included the new PMTiles support for MapLibre Native.
- The MapLibre Native Development Team kicked off their work again this January. They are currently working on:
  - Extending Custom Drawable Layer support for cross-platform custom layers.
  - Using a dynamic texture atlas for reduced memory usage.
  - Adding support for [MapLibre Tiles](https://github.com/maplibre/maplibre-tile-spec) which we expect will bring better performance.
  - Ongoing performance and stability improvements.
- We want to bring Vulkan and Metal support to Qt. Qt is currently using an older `opengl2` branch. If you are a [MapLibre Native Qt](https://github.com/maplibre/maplibre-native-qt) user and want to help, please see [this draft PR](https://github.com/maplibre/maplibre-native-qt/pull/59).
- @TatyanaPolunina, a C++ engineer from Lyft, will be addressing some stability issues Lyft is facing, well as extending the observability of MapLibre Native. We are happy to work with Tetiana and Lyft on MapLibre Native this year!
- MapLibre Native gained experimental Rust support ([docs](https://maplibre.org/maplibre-native/docs/book/rust.html), [PR](https://github.com/maplibre/maplibre-native/pull/3137)). This allows C++ code to use Rust modules including dependencies. The first demo replaces csscolorparser.hpp with a popular [Rust lib](https://crates.io/crates/csscolorparser) (2.2 million downloads). We need feedback on any potential issues with this approach. Thanks to @louwers and @nyurik.
- Work [has began](https://github.com/maplibre/maplibre-native/pull/3181) on MapLibre Native Rust wrapper - ability to use native lib as a Rust crate. This will add another supported platform, in addition to Linux/Android/iOS/Node.js/...


Copied last two points from @nyurik

**Comment by josxha** [comment]
> @josxha , does "web" here refer to this [WASM](https://docs.flutter.dev/platform-integration/web/wasm) compilation target, or a GL JS adapter? If Wasm, then I'd be interested in [documenting it](https://birkskyum.github.io/maplibre-native-wasm/flutter-wasm-target/), and analyzing the output to see how it differs from the build made by [made by Qt](https://birkskyum.github.io/maplibre-native-wasm/qt-for-webassembly/).

@birkskyum My phasing was a bit unclear, I meant that PMTiles support on web for the flutter maplibre package is planned. Web support in general is already available ([see it in action in the live demo](https://flutter-maplibre.pages.dev/demo/)).
I can't follow your message enirely. The flutter package is written so that the flutter app can get compiled to js or wasm using the [wasm compilation target](https://docs.flutter.dev/platform-integration/web/wasm) that you linked. It uses MapLibre GL JS via javascript bindings. 
It could be possible to bind to a wasm compilation of MapLibre Native but as far as I know there is no way to include mbgl directly into the app on web. I hope that somehow answers your message. 😅 I'm open to collaborate on any analyzing and documentation on that topic.

**Comment by tordans** [comment]
Cross posting https://osmus.slack.com/archives/C01G3D28DAB/p1739424395993989

**Comment by frankkienl** [comment]
Android Auto Sample app available
https://github.com/maplibre/MapLibre-Android-Auto-Sample

**Comment by nyurik** [comment]
We should have a MapLibre tile format news section, and in there put all the relevant news, i.e.
* @TimSylvester has been working on MLT decoder implementation in C++ - https://github.com/maplibre/maplibre-tile-spec/pull/441
* @jjcfrancisco and @nyurik have published [Rust version](https://crates.io/crates/fastpfor) of FastPFOR algorithm - the foundation of the MLT format.

**Comment by josxha** [comment]
## MapLibre Flutter

Version `0.2.1` of [maplibre](https://github.com/josxha/flutter-maplibre) got released this month. Key feature of this release is the possibility of synchonous controller calls on android due to the exciting release of [Flutter 3.29](https://medium.com/flutter/whats-new-in-flutter-3-29-f90c380c2317). This is a big milestone to seemlessly display and position Flutter widgets on top of the map. Check out the [changelog](https://github.com/josxha/flutter-maplibre/blob/main/CHANGELOG.md#021) for all other improvements.

[Support for iOS](https://github.com/josxha/flutter-maplibre/issues/26) made progress as well with its completion hopefully in March. 

<table>
<tr>
<td>Flutter widgets on top of the map view</td>
<td>iOS support preview</td>
</tr>
<tr>
<td><video src="https://github.com/user-attachments/assets/b1adfccb-c3b7-45cc-9892-0b82becf8c20"></td>
<td><video src="https://github.com/user-attachments/assets/aee5b7cc-9648-4b9f-aa7f-8c0ae8aecf7a"></video></td>
</tr>
</table>

**Comment by HarelM** [comment]
## Web

This was a bit of a short month so in this month we only released a single version: 5.1.1.
`addProtocol` showed again its power and a plugin was released to support arbitrary fonts loading using tinySDF: https://www.npmjs.com/package/maplibre-local-glyphs, this will hopefully pave the path to other font functionality that's coming in the near future.

**Comment by louwers** [comment]
Sorry for the delay!

## MapLibre Native

- Implementation of the new [MapLibre Tiles format](https://github.com/maplibre/maplibre-tile-spec#maplibre-tiles-mlt) is in full swing. There is a [PR in the `maplibre/maplibre-tile-spec`](https://github.com/maplibre/maplibre-tile-spec/pull/441) repo with a C++ reader implementation and a [draft PR](https://github.com/maplibre/maplibre-native/pull/3246) for the intergration in MapLibre Native.

  Even though development has only recently started, this proof-of-concept is already able to use MLTs (MapLibre Tiles) as a source alongside [MVTs](https://github.com/mapbox/vector-tile-spec).
  ![](https://github.com/user-attachments/assets/5eaf708a-3f0b-4f6e-a800-aec732bff530)
  *Screenshot showing the road network in Europe from some demo MLTs based on Bing Maps data*.

- [MapLibre iOS 6.11.0](https://github.com/maplibre/maplibre-native/releases/tag/ios-v6.11.0) was released.
	- Custom Style Layers, allow drawing on the map using the Metal API. It can now be used from Swift. See [this article](https://maplibre.org/maplibre-native/ios/latest/documentation/maplibre-native-for-ios/customstylelayerexample) in the documentation.
    - In the upcoming MapLibre iOS 6.12.0 release we are exposing some low-level events for better observability, see [this article](https://maplibre.org/maplibre-native/ios/latest/documentation/maplibre-native-for-ios/observerexample).

- Custom Drawable Layers allow drawing on the map with in a platform-independent way. Currently you need to use C++ if you want to use custom drawable layers. This month [a PR](https://github.com/maplibre/maplibre-native/pull/3210) was merged that adds some additional functionality and improvements to custom drawable layer, among other things a way to upload (3D) geometries.
  ![](https://github.com/user-attachments/assets/735e0c3f-a68e-4e99-856a-26bfee6cccc6)
  *Custom Drawable Layer demo available in the GLFW app if you press 'V'.*

**Comment by samuelvdv** [comment]
Hi, 

in a previous newsletter I saw the request for projects to spotlight. I would like to propose our project for spotlight that is extensively using MapLibre for visualising indoor & outdoor positioning within our Pozyx platform. Indoor maps can be uploaded using the IMDF format which is used to show the 3D indoor map.

https://github.com/user-attachments/assets/a3f519e8-c248-4ce8-901c-ab52b099d23b

Apart from showing assets on a map, we also visualise map-related analytics like heamaps and bar charts. We've also used a lot of the drawing functionality for creating and editing geofences.

![Image](https://github.com/user-attachments/assets/d4442b85-a272-479d-a403-7fce6d527ea2)
![Image](https://github.com/user-attachments/assets/77d7e154-808f-4072-8466-a28e8a080c77)

More information here: https://www.pozyx.io/products/rtls-platform 
We've been using MapLibre in our product for over 3 years now and it has been a super good choice for us. Haven't regretted it once!

Thanks, 
Samuel Van de Velde
CTO [Pozyx](https://www.pozyx.io/)

**Comment by MubarakNative** [comment]
Medium article getting started with maplibre native

https://github.com/maplibre/maplibre-native/discussions/3289

**Comment by HarelM** [comment]
## Web

We have release two minor versions this month: 5.2.0 and 5.3.0.
The content of these versions is mainly bug-fixes and small features.
Supporting fonts by using system fonts instead of the `glyphs` field is underway and so is the global map state.
These awesome features should land in future releases allowing style authors to focus on how the map would look and be able to theme it more easily, so stay tuned!
Thanks for all the contributors for making all these adjustments and improvements!

**Comment by louwers** [comment]
## MapLibre Native

- [Grab](https://en.wikipedia.org/wiki/Grab_Holdings) made some [contributions](https://github.com/maplibre/maplibre-native/pulls?q=author%3Agrab-liyanjin) this month. They are currently using an internal fork but will start using MapLibre Native this year. We are looking collaborate with Grab and help them on board.
- MapLibre iOS [6.12.1](https://github.com/maplibre/maplibre-native/releases/tag/ios-v6.12.1) was released.
- MapLibre Android [11.8.3](https://github.com/maplibre/maplibre-native/releases/tag/android-v11.8.3), [11.8.4](https://github.com/maplibre/maplibre-native/releases/tag/android-v11.8.4) and [11.8.5](https://github.com/maplibre/maplibre-native/releases/tag/android-v11.8.5) were released.
- We started making releases of the core as a [static library](https://github.com/maplibre/maplibre-native/releases?q=core-&expanded=true). There are no API stability guarantees right now. This is mainly to facilitate the [Rust wrapper](https://crates.io/crates/maplibre_native) for MapLibre Native that is under development. If you have some kind of use case for this [let us know](https://github.com/maplibre/maplibre-native/discussions/categories/show-and-tell) about it.
- Some neat improvements to look out for that are in the pipeline:
  - [Level-of-detail controls](https://github.com/maplibre/maplibre-native/pull/2958), useful when you need to limit the amount of tiles that are loaded such as during navigation.
  - A [rendering HUD](https://github.com/maplibre/maplibre-native/pull/3322) you can toggle to see statistics relevant for performance at a glance. Especially useful for users of MapLibre Native on embedded hardware where getting these numbers through other means may be difficult.
  - Support for [MapLibre Tiles](https://github.com/maplibre/maplibre-native/pull/3246) with advantages such as better performance (better battery life) and lower bandwidth.

<img src="https://github.com/user-attachments/assets/ed90e5d8-f24e-425a-9e25-7e3f956074fa" width="50%" />

_Rendering HUD screenshot_

**Comment by ramyaragupathy** [comment]
For upcoming newsletter from Oliver:

_With the maplibre-gl-complex-text plugin you can render South and South-East Asian text in MapLibre GL JS.
Supports Bengali, Devanagari, Gujarati, Gurmukhi, Kannada, Khmer, Malayalam, Myanmar, Oriya, Tamil, and Telugu.
Since I cannot read any of these languages I am not completely sure if it works as expected. Let me know if you find any bugs...
https://github.com/wipfli/maplibre-gl-complex-text/_

Interactive map: https://wipfli.github.io/maplibre-gl-complex-text/swiss-map?lang=name%3Ahi#map=3.57/25.06/61.76

**Comment by louwers** [comment]
## MapLibre Native

> [!WARNING]
> Work in Progress

 https://github.com/maplibre/maplibre-native/discussions/3414

**Comment by oxcar** [comment]
Hello everybody,

I wanted to share a project that uses MapLibre.

[https://rutaton.zamora.gob.mx/](https://rutaton.zamora.gob.mx/)

<img width="2672" alt="Image" src="https://github.com/user-attachments/assets/005952c3-e06f-4b68-9f92-dc71083ad6f7" />

It's a GTFS visualization for the Zamora public transport network. It's part of a crowdmapping project where we helped the city digitize their routes with help of local students and using mobile phones. The objective was to create the information so citizens hava access to this information that was no previously available and also to make it available to other interested parties.

I developed it using Svelte + MapLibre and it was super easy to get something done quickly.

Saludos 👋

Óscar Hernández
Tech Lead + Projects @ Codeando México

**Comment by HarelM** [comment]
## Web

We have released two versions this month, a minor version 5.3.1 that incorporated some bug fixes and 5.4 a major version which included a way to control the level of details (LOD) for different pitch and zoom levels. 
Map-state and more options for hillshade are coming soon, so there's a lot to expect! 
There was also an initial development of a plugin that support complicated scripting languages as mentioned in previous posts. 
Be sure to checkout awesome-maplibre repo to see all kind of amazing work out community is doing!

**Comment by CommanderStorm** [comment]
# Martin

We are currently discussing what features/changes we want to do before we can release martin 1.0.
If you have ideas/oppinons what we should/need to do, we would appreciate your oppinion here: https://github.com/maplibre/martin/discussions/1805

**Comment by louwers** [comment]
Thanks @ramyaragupathy!

New issue for May 👉 https://github.com/maplibre/maplibre.github.io/issues/420

**Comment by onuryb** [comment]
Maplibre sky is not displayed everywhere is covered with maps @HarelM
